gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+            {
+               uniform.offset = src.ubo_offset;
+               uniforms[SLANG_CBUFFER_UBO].push_back(uniform);
+            }
+         }
+      }
+      texture_map++;
+   }
+
+   out->texture_count = textures.size();
+
+   textures.push_back({ NULL });
+   out->textures = (texture_sem_t*)malloc(textures.size() * sizeof(*textures.data()));
+   memcpy(out->textures, textures.data(), textures.size() * sizeof(*textures.data()));
+
+   for (int i = 0; i < SLANG_CBUFFER_MAX; i++)
+   {
+      if (uniforms[i].empty())
+         continue;
+
+      out->cbuffers[i].uniform_count = uniforms[i].size();
+
+      uniforms[i].push_back({ NULL });
+      out->cbuffers[i].uniforms =
+            (uniform_sem_t*)malloc(uniforms[i].size() * sizeof(*uniforms[i].data()));
+      memcpy(
+            out->cbuffers[i].uniforms, uniforms[i].data(),
+            uniforms[i].size() * sizeof(*uniforms[i].data()));
+   }
+
+   return true;
+}
+
+bool slang_process(
+      video_shader*          shader_info,
+      unsigned               pass_number,
+      enum rarch_shader_type dst_type,
+      unsigned               version,
+      const semantics_map_t* semantics_map,
+      pass_semantics_t*      out)
+{
+   Compiler*          vs_compiler = NULL;
+   Compiler*          ps_compiler = NULL;
+   video_shader_pass& pass        = shader_info->pass[pass_number];
+   glslang_output     output;
+
+   if (!glslang_compile_shader(pass.source.path, &output))
+      return false;
+
+   if (!slang_preprocess_parse_parameters(output.meta, shader_info))
+      return false;
+
+   pass.source.string.vertex   = NULL;
+   pass.source.string.fragment = NULL;
+
+   try
+   {
+      ShaderResources vs_resources;
+      ShaderResources ps_resources;
+      string          vs_code;
+      string          ps_code;
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         vs_compiler = new CompilerHLSL(output.vertex);
+         ps_compiler = new CompilerHLSL(output.fragment);
+      }
+      else
+      {
+         vs_compiler = new CompilerGLSL(output.vertex);
+         ps_compiler = new CompilerGLSL(output.fragment);
+      }
+
+      vs_resources = vs_compiler->get_shader_resources();
+      ps_resources = ps_compiler->get_shader_resources();
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         CompilerHLSL::Options options;
+         CompilerHLSL*         vs = (CompilerHLSL*)vs_compiler;
+         CompilerHLSL*         ps = (CompilerHLSL*)ps_compiler;
+         options.shader_model     = version;
+         vs->set_options(options);
+         ps->set_options(options);
+
+         std::vector<HLSLVertexAttributeRemap> vs_attrib_remap;
+#if 0
+         /* remaps vertex shader output too so disable for now */
+         vs_attrib_remap.push_back({ 0, "POSITION" });
+         vs_attrib_remap.push_back({ 1, "TEXCOORD" });
+#endif
+         /* not exactly a vertex attribute but this remaps
```

Comment:
This doesn't make sense. Why is FragCoord mapped to anything?

Reply:
some shaders use it...
unless I add the SV_POSITION semantic to it the linker will complain about an input attribute not being provided by the vertex shader.

Reply:
SV_Position must be provided by the vertex shader, that's gl_Position. Do you have an example?

Reply:
https://github.com/libretro/slang-shaders/blob/master/xsal/shaders/2xsal-level2.slang
looks like it is just an unused input afterall heh. 
the hlsl compiler would still fail without the remapping though since the code starts by assigning it to the static variable.
```...
static float4 FragCoord;
...
SPIRV_Cross_Output main(SPIRV_Cross_Input stage_input)
{
...
    FragCoord = stage_input.FragCoord;
    frag_main();
...
}
```

SV_Position is gl_Position when used as vertex shader output and gl_FragCoord when used  as pixel shader input, yes, a bit unusual (don't ask me xD).